onnx2kerastl/pooling_layers.py

```diff
@@ -211,23 +212,30 @@ def convert_topk(node, params, layers, lambda_func, node_name, keras_name):
     to_sort = bool(params.get('sorted', 1))
     x = layers[node.input[0]]
     k = layers[node.input[1]][0]
-    if not is_numpy(k) and not K.is_keras_tensor(k): # Eager tensor does not serialize well
+    if not is_numpy(k) and not K.is_keras_tensor(k):  # Eager tensor does not serialize well
         k = k.numpy().astype(np.int32)
     if not largest:
         in_tensor = -x
     else:
         in_tensor = x
 
-    def target_layer(in_tensor, k=k, to_sort=to_sort, axis=axis):
+    def target_layer(composed_input, to_sort=to_sort, axis=axis):
+        in_tensor = composed_input[..., :-1]
+        k = composed_input[..., -1]
+        k = tf.cast(tf.reshape(k, [1])[0], tf.int32)
         rank = len(in_tensor.shape)
+
         if axis >= rank - 1 or axis == -1:
             permuted = in_tensor
         else:
             ord_permute = np.arange(rank)
             ord_permute[axis] = rank - 1
             ord_permute[-1] = axis
             permuted = tf.transpose(in_tensor, ord_permute)
-        if (isinstance(k, tf.Tensor) or (not isinstance(k, (np.ndarray, np.generic)) and not isinstance(k, int) and tf.keras.backend.is_keras_tensor(k)))\
+
+        if (isinstance(k, tf.Tensor) or (not isinstance(k, (np.ndarray, np.generic)) and not isinstance(k,
```

Comment:
I think this casting block is redundent now. since you always cast on the top, this shouldn't change anything. Consider removing?